### DISCUSSION TITLE
question about Initialization of the entropy model

### DISCUSSION THREAD

**Discussion by AlbertoPresta** [InterDigitalInc/CompressAI#133] [Q&A]
Hello, 
I have a question related to the entropy model and the way you optimize the range of the likelihood.
Why did you initialize the tail_mass to 1e-9, and doing so the target around 21.4? is there any reason to do this? is there any documentation/paper with all these kinds of particular features? 

thanks in advance for the answer, 
Alberto

**Comment by fracape** [comment]
Hi, thank you for spotting this. Most initial parameters in CompressAI are directly derived from the numbers reported by the authors, or from their available source code. As for the tail_mass, tensorflow-compression uses 2**-8. I think we made a mistake here and used the value of the likelihood bound for tail_mass as well. So you are right, we get a target of ~21.4 instead of ~6.24, which means less values outside the range and Golomb coded. Going to run some experiments to check the impact.

**Reply by yykzyj123456** [reply]
> Hi, thank you for spotting this. Most initial parameters in CompressAI are directly derived from the numbers reported by the authors, or from their available source code. As for the tail_mass, tensorflow-compression uses 2**-8. I think we made a mistake here and used the value of the likelihood bound for tail_mass as well. So you are right, we get a target of ~21.4 instead of ~6.24, which means less values outside the range and Golomb coded. Going to run some experiments to check the impact.

Hello,
I am also very confused about the meaning of 'tail_mass' and 'target'?  If entropy encoding is not performed, and only the likelihood is used to estimate the encoding length, can aux_loss not be used?
thanks in advance for the answer,
yyk
